testdata/gapi/onnx/download_onnx_models.py

```diff
@@ -0,0 +1,152 @@
+            return self.sha == sha.hexdigest()
+        except Exception as e:
+            print('[Excn]: Catch {}'.format(e))
+
+    def create_dir(self):
+        model_dir_path = self.MODELS_DIR + self.filepath
+        if not os.path.exists(model_dir_path):
+            os.makedirs(model_dir_path)
+        else:
+            print('[Warn]: Directory already contains a folder for {} - skipping'.format(self.name))
+
+    def download(self):
+        print('______________________{}______________________'.format(self.name))
+        print('[Info]: Create directory for {}'.format(self.name))
+        self.create_dir()
+        if self.verify():
+            print('[Warn]: Hash match - skipping')
+            return True
+        print('[Info]: Download model')
+        model_cache_file_path = CACHE_DIR + self.filepath + self.name + '_cache.onnx'
+        model_cache_model_path = CACHE_DIR + self.model_path
+        # Copy cache file with new name
+        if not os.path.exists(model_cache_file_path):
+            os.replace(model_cache_model_path, model_cache_file_path)
+        # Pull large model file
+        os.system(self.sys_lfs_call())
+        # Move large model file to onnx_models
+        os.replace(model_cache_model_path, self.MODELS_DIR + self.model_path)
+        # Rename cache file to old name
+        os.rename(model_cache_file_path, model_cache_model_path)
+        return self.verify()
+
+def download_cache():
+    print('______________________{}______________________'.format('Download cache'))
+    if not os.path.exists(CACHE_DIR):
+        print('[Info]: Cloning onnx_models repository from https://github.com/onnx/models.git')
+        os.system('git clone --recursive https://github.com/onnx/models.git onnx_models_cache')
+    else:
+        print('[Warn]: Directory already contains a "onnx_models_cache" folder - skipping')
+
+models = [
+    Model(
+        name ='ssd_mobilenet_v1_10',
+        filepath ='vision/object_detection_segmentation/ssd-mobilenetv1/model/',
+        sha = '6a2ba88990166b5212fc4115bb347dd1402fbf39'
+    ),
+    Model(
+        name='squeezenet1.0-9',
+        filepath='vision/classification/squeezenet/model/',
+        sha = '7c4a0cc990d877f46105eb331bb71e2c90c0ecbb'
+    ),
+    Model(
+        name='emotion-ferplus-8',
+        filepath='vision/body_analysis/emotion_ferplus/model/',
+        sha = '073ea68e09c0c8c21401b95e9cdccb42c639bc75'
+    ),
+    Model(
+        name='FasterRCNN-10',
+        filepath='vision/object_detection_segmentation/faster-rcnn/model/',
+        sha = '7df2f48a6429ea412733af8ce9673a092a5f84c4'
+    ),
+    Model(
+        name='yolov3-10',
+        filepath='vision/object_detection_segmentation/yolov3/model/',
+        sha = 'a3e31b46f37c2b5de0fc85b6b54571898e7bbbb7'
+    ),
+    Model(
+        name='tinyyolov2-8',
+        filepath='vision/object_detection_segmentation/tiny-yolov2/model/',
+        sha = '7ad8395edc8057030d17c14459de6d07f4d11ac6'
```

Comment:
So, the script will expect exactly these versions of models, right?
But what if a model's updated so its sha's changed - it would require to update this script, these hardcoded shas?

Maybe, there's a way to collect models' current sha versions from somewhere? Or it is not so important, is it?

Reply:
No. Hardcode. OCV script uses the same way.

Reply:
Somebody would say we should tend to surpass our ancestors, but as you wish